src/compiler/checker/lines_16000_18000.rs

```diff
@@ -132,6 +140,195 @@ impl TypeChecker {
         }
     }
 
+    pub(super) fn create_type_mapper(
+        &self,
+        sources: Vec<Rc<Type /*TypeParameter*/>>,
+        targets: Option<Vec<Rc<Type>>>,
+    ) -> TypeMapper {
+        if sources.len() == 1 {
+            self.make_unary_type_mapper(
+                sources[0].clone(),
+                if let Some(targets) = targets {
+                    targets[0].clone()
+                } else {
+                    self.any_type()
+                },
```

Comment:
I'd usually use `map_or_unwrap` for things like this.

Reply:
Ah I think I'm used to reaching for `.unwrap_or()`/`.unwrap_or_else()` but not `.map_or()`/`.map_or_else()`, updated to use `.map_or_else()`